code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,648 @@
+					candidate.emp_act(EMP_HEAVY)
+					var/construct_class = alert(user, "Please choose which type of construct you wish to create.",,"Juggernaut","Wraith","Artificer")
+					if(QDELETED(candidate))
+						channeling = FALSE
+						return
+					user.visible_message("<span class='danger'>The dark cloud receedes from what was formerly [candidate], revealing a\n [construct_class]!</span>")
+					switch(construct_class)
+						if("Juggernaut")
+							makeNewConstruct(/mob/living/simple_animal/hostile/construct/armoured, candidate, user, 0, T)
+						if("Wraith")
+							makeNewConstruct(/mob/living/simple_animal/hostile/construct/wraith, candidate, user, 0, T)
+						if("Artificer")
+							makeNewConstruct(/mob/living/simple_animal/hostile/construct/builder, candidate, user, 0, T)
+					SEND_SOUND(user, sound('sound/magic/cult_spell.ogg', 0, 1, 25))
+					uses--
+					candidate.mmi = null
+					qdel(candidate)
+					channeling = FALSE
+				else
+					channeling = FALSE
+					candidate.color = prev_color
+					return
+			else
+				uses--
+				to_chat(user, "<span class='warning'>A dark cloud emanates from you hand and swirls around [candidate] - twisting it into a construct shell!</span>")
+				new /obj/structure/constructshell(T)
+				SEND_SOUND(user, sound('sound/magic/cult_spell.ogg', 0, 1, 25))
+				qdel(candidate)
+		else if(istype(target,/obj/machinery/door/airlock))
+			channeling = TRUE
+			playsound(T, 'sound/machines/airlockforced.ogg', 50, TRUE)
+			do_sparks(5, TRUE, target)
+			if(do_after(user, 50, target = user))
+				if(QDELETED(target))
+					channeling = FALSE
+					return
+				target.narsie_act(TRUE)
+				uses--
+				user.visible_message("<span class='warning'>Black ribbons suddenly emanate from [user]'s hand and cling to the airlock - twisting and corrupting it!</span>")
+				SEND_SOUND(user, sound('sound/magic/cult_spell.ogg', 0, 1, 25))
+				channeling = FALSE
+			else
+				channeling = FALSE
+				return
+		else
+			to_chat(user, "<span class='warning'>The spell will not work on [target]!</span>")
+			return
+		..()
+
+//Armor: Gives the target a basic cultist combat loadout
+/obj/item/melee/blood_magic/armor
+	name = "Arming Aura"
+	desc = "Will equipt cult combat gear onto a cultist on contact."
+	color = "#33cc33" // green
+
+/obj/item/melee/blood_magic/armor/afterattack(atom/target, mob/living/carbon/user, proximity)
+	if(iscarbon(target) && proximity)
+		uses--
+		var/mob/living/carbon/C = target
+		C.visible_message("<span class='warning'>Otherworldly armor suddenly appears on [C]!</span>")
+		C.equip_to_slot_or_del(new /obj/item/clothing/under/color/black(user), slot_w_uniform)
+		C.equip_to_slot_or_del(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
+		C.equip_to_slot_or_del(new /obj/item/storage/backpack/cultpack(user), slot_back)
+		C.equip_to_slot_or_del(new /obj/item/clothing/shoes/cult(user), slot_shoes)
+		if(C == user)
+			qdel(src) //Clears the hands
```

Comment:
does this work? Seeing a qdel on the src should stop the proc

Reply:
It does